src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,94 @@
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IShellHost _shellHost;
+        private readonly ShellSettings _shellSettings;
+
+        public GCCollabAuthenticationSettingsDisplayDriver(
+            IAuthorizationService authorizationService,
+            IDataProtectionProvider dataProtectionProvider,
+            IHttpContextAccessor httpContextAccessor,
+            IShellHost shellHost,
+            ShellSettings shellSettings)
+        {
+            _authorizationService = authorizationService;
+            _dataProtectionProvider = dataProtectionProvider;
+            _httpContextAccessor = httpContextAccessor;
+            _shellHost = shellHost;
+            _shellSettings = shellSettings;
+        }
+
+        public override async Task<IDisplayResult> EditAsync(GCCollabAuthenticationSettings settings, BuildEditorContext context)
+        {
+            var user = _httpContextAccessor.HttpContext?.User;
+            if (user == null || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageGCCollabAuthentication))
+            {
+                return null;
+            }
+
+            return Initialize<GCCollabAuthenticationSettingsViewModel>("GCCollabAuthenticationSettings_Edit", model =>
+            {
+                model.ClientID = settings.ClientID;
+                if (!string.IsNullOrWhiteSpace(settings.ClientSecret))
+                {
+                    //map to a magic string so the api key is not leaked in the html
+                    model.ClientSecret = MagicString;
+                }
+                else
+                {
+                    model.ClientSecret = string.Empty;
+                }
+                if (settings.CallbackPath.HasValue)
+                {
+                    model.CallbackUrl = settings.CallbackPath.Value;
+                }
+            }).Location("Content:5").OnGroup(GCCollabConstants.Features.GCCollabAuthentication);
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(GCCollabAuthenticationSettings settings, BuildEditorContext context)
+        {
+            if (context.GroupId == GCCollabConstants.Features.GCCollabAuthentication)
+            {
+                var user = _httpContextAccessor.HttpContext?.User;
+                if (user == null || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageGCCollabAuthentication))
+                {
+                    return null;
+                }
+
+                var model = new GCCollabAuthenticationSettingsViewModel();
+                await context.Updater.TryUpdateModelAsync(model, Prefix);
+
+                if (context.Updater.ModelState.IsValid)
+                {
+                    var protector = _dataProtectionProvider.CreateProtector(GCCollabConstants.Features.GCCollabAuthentication);
+
+                    settings.ClientID = model.ClientID;
+                    if(!(model.ClientSecret == MagicString))
```

Comment:
model.ClientSecret != MagicString

Reply:
Maybe a blank check as well, unless the protector already handles it.

Reply:
Not sure if data protector handles it or not I just checked if it's empty.